src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -29,31 +29,57 @@ Copyright (c) .NET Foundation. All rights reserved.
     included in the generated deps.json.
     ============================================================
     -->
+  <UsingTask TaskName="FindNativeDeps" AssemblyFile="$(ILLinkTasksAssembly)" />
   <Target Name="_ILLink"
           Condition=" '$(PublishTrimmed)' == 'true' And
                       '$(_TargetFrameworkVersionWithoutV)' >= '3.0' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' "
           DependsOnTargets="_RunILLink">
 
+    <ItemGroup>
+      <_LinkedManagedAssemblies Include="@(_LinkedManagedAssembliesCandidates)" Condition="Exists('%(Identity)')" />
+    </ItemGroup>
+
+    <!-- Filter out native assets no longer needed after linking. -->
+    <ItemGroup Condition=" '$(TrimmerKeepNativeAssets)' != 'true' ">
+      <_NativeAssetsToAlwaysKeep Include="coreclr.dll;libcoreclr.dylib;libcoreclr.so" />
+      <_NativeAssetsToAlwaysKeep Include="clrjit.dll;libclrjit.dylib;libclrjit.so" />
+      <_NativeAssetsToAlwaysKeep Include="hostfxr.dll;libhostfxr.dylib;libhostfxr.so" />
+      <_NativeAssetsToAlwaysKeep Include="hostpolicy.dll;libhostpolicy.dylib;libhostpolicy.so" />
+      <!-- The full publish deps file path is generated after this target, so just depend on the filename. -->
+      <_NativeAssetsToAlwaysKeep Include="$(ProjectDepsFileName)" />
+      <_NativeAssetsToAlwaysKeep Include="$(ProjectRuntimeConfigFilePath)" />
```

Comment:
Why is adding the deps file necessary here?  Does FindNativeDeps recognize and only filter native binaries? And leave away other /text files?

Reply:
Yes, it only recognizes native binaries. Adding the deps file here is a workaround for the lack of a proper input set that represents the set of runtime assemblies. The input `_NativeAssetsToFilter` is computed as `ResolvedFileToPublish`-`_ManagedAssembliesToLink`, so it includes text files and other things. The out-of-band linker package used `ResolvedAssembliesToPublish`, which was a closer approximation. For this to work with apps that publish additional content files, we'll need to compute the correct input set.

Reply:
I agree that we should compute and use the correct input set.
In the meantime, the task should keep any non-native binary file fed into it.
I think being conservative about other file types is better than special-casing just the json configuration files.